read-the-docs-2020-stats.rst

```diff
@@ -0,0 +1,117 @@
+.. post:: Jan 14, 2021
+   :tags: stats, year-in-review
+
+Read the Docs 2020 Stats
+========================
+
+2020 was a rough year for everyone,
+including our team.
+We managed to make it through,
+and continue to have 5 folks working full-time to make Read the Docs better for you.
+
+We are going into 2021 with a :doc:`new grant </czi-grant-announcement>`,
+which will require us to do some hiring.
+We also launched our `EthicalAds network`_,
+which is bringing our approach to sustainability to the tech community as a whole.
+
+As we have for the past 7 years,
+we are proud to share our 2020 stats post.
+
+.. _EthicalAds network: https://www.ethicalads.io/
+
+.. note::
+
+	You can always see our stats for the last `30 days`_.
+
+	Our posts from 2013_, 2014_, 2015_, 2016_, 2017_, 2018_, and 2019_ are also available.
+
+.. _Read the Docs: https://readthedocs.org/
+.. _30 days: http://www.seethestats.com/site/readthedocs.org
+.. _2013: https://blog.readthedocs.com/read-the-docs-2013-stats/
+.. _2014: https://blog.readthedocs.com/read-the-docs-2014-stats/
+.. _2015: https://blog.readthedocs.com/read-the-docs-2015-stats/
+.. _2016: https://blog.readthedocs.com/read-the-docs-2016-stats/
+.. _2017: https://blog.readthedocs.com/read-the-docs-2017-stats/
+.. _2018: https://blog.readthedocs.com/read-the-docs-2018-stats/
+.. _2019: https://blog.readthedocs.com/read-the-docs-2019-stats/
+
+
+Page Views
+----------
+
+Our stats:
+
+* 580 Million Page Views (from 465 million)
+* 156 Million Unique Visitors (from 120 million)
+
+.. From Google Analytics
+
+These numbers are not completely accurate, since we :doc:`implemented Do Not Track support </do-not-track>` our analytics pageview figures understate our actual traffic.
+We estimate that around 6% of users are not counted here.
+We also don't count all the users who block Google Analytics with a browser extension.
+
+Site Stats
+----------
+
+The stats, in total numbers:
+
+* 240,000 projects (from 200,000)
+* 430,000 users (from 400,000)
+* 2,480,063 builds (from 10 million total last year)
+
+We have been battling spam quite heavily again this year,
+so these numbers are a bit skewed.
+We do notice a uptick in usage across the site,
+both in terms of support and traffic.
+
+.. Project.objects.count()
+.. User.objects.count()
+.. Build.objects.filter(date__year__lte=2021).first().pk - Build.objects.filter(date__year__lte=2020).first().pk
+
+Community
+---------
+
+This year, we had:
+
+* 14 `people`_ who committed code to the main repository (from 21)
+* 3636 `commits`_ (from 4793)
+* 532 `issues`_ - 104 open, 428 closed (from 609)
+
+We have been working to move user support issues to email from GitHub,
+which likely addresses the reduction in ticket numbers.
+We are also working to expand our ecosystem of projects,
+so the commits to the main repository are down.
```

Comment:
Yea, I just wanted to keep the metrics the same over time. It seems annoying to calculate this, and the repos we care about presumably change over time?

Reply:
Yeah, but if we are going to start distributing our work between .org, theme, sphinx extensions, an other important repos, I think it worth start considering this. In particular this year, 2021 that we have the grant and more people involved.

So, if we can start saving the 2020 values for this now, I think it will be good data by the end of this year.

Reply:
@humitos do you want to compile that data? The file has how I normally do this, but we'd need to diff the users across repos somehow.

Reply:
I think this may work for number of commits and number of contributors

```
~/rtfd/code                                                                                                                                                                                                  
▶ for repository in blog ethical-ad-client ethical-ad-server readthedocs-docker-images readthedocs-sphinx-ext readthedocs.org recommonmark sphinx-hoverxref sphinx-autoapi sphinx-notfound-page sphinx_rtd_theme; do echo -n "${repository}: " && cd ${repository} && git rev-list --count --all --after="2019-12-31" --before="2021-01-01" && git log --after="2019-12-31" --before="2021-01-01" --pretty="%an" | sort | uniq > committers.txt && cd ..; done ; echo -n "People: " && cat */committers.txt | sort | uniq | wc -l
blog: 70
ethical-ad-client: 93
ethical-ad-server: 457
readthedocs-docker-images: 174
readthedocs-sphinx-ext: 48
readthedocs.org: 2668
recommonmark: 3
sphinx-hoverxref: 135
sphinx-autoapi: 83
sphinx-notfound-page: 84
sphinx_rtd_theme: 222
People: 47
```

I got the number of issues from GH https://github.com/search?q=org%3Areadthedocs+is%3Aissue+created%3A2020-01-01..2020-12-31

- Closed: 645
- Open: 310

Reply:
@humitos This doesn't include number of contributors though right? We need to combine them, then dedupe them, then count them. Seemed...annoying.

Reply:
Contributors are included there, `People: 47`. There are 1 `git log ... > commiters.txt` command + `sort | uniq` and then a `cat */commiters.txt | sort | uniq | wc -l`. Is that what we wanted or do we want contributors per-project?

Reply:
Just a note, I have been squashing PRs before merging lately, so that could also be the reason for fewer commits :)

Reply:
@humitos Gotcha, nice! 👍